### DISCUSSION TITLE
Support multi test case relationships

### DISCUSSION THREAD

**Discussion by thomaswong-pixieset** [DamianOsipiuk/testcafe-reporter-testrail#37] [Ideas]
Hi!

My team and I currently have a subset of automated tests that have more than a 1-to-1 relationship with our test cases. We were wondering if multi CID support is something that might be potentially added in the future.

In theory, test results will be published to both test cases in TestRail regardless of the result. 

```
test.meta({ CID: 'C123' }, { CID: '124' })(
	'Verify Homepage Functionality', async (t) => { ... });
```

**Comment by DamianOsipiuk** [comment]
Hi @thomaswong79.
So adding ability to provide multiple `CID's`  should be pretty easy.
It would look like `test.meta({ CID: ['C123', 'CID124']})`.

There are some constrains though, due to how data reporting and configuration is done right now:
- both test cases need to exist in the same project
- both test cases need to exist in the same suite

Considering both of the above does it make sense to cover multiple test cases in the same suite with one test?
If they are tested by the same piece of code they would always fail/pass together, so from my point of view, it would make more sense to merge those two test cases into one.

**Comment by thomaswong-pixieset** [comment]
Thanks for your quick response @DamianOsipiuk! 

Yes, ideally both of your bullet points are fulfilled with our current setup. The main reason behind multiple test cases being run in a singular test is due to speed (many test cases follow a similar user flow, thus allowing us to "test" multiple features and functions "while we're there". 

As we write new automated tests, they will be on a 1-to-1 basis for better disassociation for test results however, we have a bunch of old tests that have not been refactored out yet and would love for the support to be added. 😃

**Comment by TAkbay** [comment]
I would love to have this feature too :D

**Comment by thomaswong-pixieset** [comment]
Added PR https://github.com/DamianOsipiuk/testcafe-reporter-testrail/pull/39 in reference to the above. I'm very novice in TS, so let me know if I can make any modifications.
